src/psyclone/domain/lfric/kernel_interface.py

```diff
@@ -0,0 +1,793 @@
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        '''
+        fs_name = function_space.orig_name
+        undf_symbol = self._create_symbol(
+            "undf_{0}".format(fs_name),
+            lfric_psyir.NumberOfUniqueDofsDataSymbol, extra_args=[fs_name])
+        self._arglist.append(undf_symbol)
+
+        fs_name = function_space.orig_name
+        ndf_symbol = self._create_symbol(
+            "ndf_{0}".format(fs_name), lfric_psyir.NumberOfDofsDataSymbol,
+            extra_args=[fs_name])
+
+        dofmap_symbol = self._create_symbol(
+            "dofmap_{0}".format(fs_name), lfric_psyir.DofMapDataSymbol,
+            dims=[ndf_symbol], extra_args=[fs_name])
+        self._arglist.append(dofmap_symbol)
+
+    def banded_dofmap(self, function_space, var_accesses=None):
+        '''Not implemented.
+
+        :param function_space: the function space for this dofmap.
+        :type function_space: :py:class:`psyclone.domain.lfric.FunctionSpace`
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NotImplementedError: as this method is not implemented.
+
+        '''
+        raise NotImplementedError("banded_dofmap not implemented")
+
+    def indirection_dofmap(self, function_space, operator=None,
+                           var_accesses=None):
+        '''Not implemented.
+
+        :param function_space: the function space for this dofmap.
+        :type function_space: :py:class:`psyclone.domain.lfric.FunctionSpace`
+        :param operator: the CMA operator.
+        :type operator: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NotImplementedError: as this method is not implemented.
+
+        '''
+        raise NotImplementedError("indirection_dofmap not implemented")
+
+    def basis(self, function_space, var_accesses=None):
+        '''Create an LFRic basis function argument and add it to the symbol
+        table and argument list.
+
+        :param function_space: the function space for this basis function.
+        :type function_space: :py:class:`psyclone.domain.lfric.FunctionSpace`
+        :param var_accesses: an unused optional argument that stores \
+            information about variable accesses.
+        :type var_accesses: :\
+            py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        '''
+        basis_name_func = function_space.get_basis_name
+        from psyclone.dynamo0p3 import DynBasisFunctions
```

Comment:
Please disable the pylint warning for this import (and the one below) and add comments to say they are here to avoid circular deps.

Reply:
Yes, good point. Done.